lib/commands/audit.js

```diff
@@ -188,19 +190,41 @@ class VerifySignatures {
   }
 
   async setKeys ({ registry }) {
-    const keys = await fetch.json('/-/npm/v1/keys', {
-      ...this.npm.flatOptions,
-      registry,
-    }).then(({ keys: ks }) => ks.map((key) => ({
-      ...key,
-      pemkey: `-----BEGIN PUBLIC KEY-----\n${key.key}\n-----END PUBLIC KEY-----`,
-    }))).catch(err => {
-      if (err.code === 'E404' || err.code === 'E400') {
-        return null
-      } else {
-        throw err
-      }
-    })
+    const { host, pathname } = new URL(registry)
+    const regKey = `${host}${pathname === '/' ? '' : pathname}/keys.json`
```

Comment:
Should this also use a `path.join` to handle any trailing slashes in the path?
```suggestion
    const regKey = path.join(host, pathname, "keys.json")
```

Reply:
I think `path.join` isn't recommended when constructing URLs (don't wanna end-up with `/` instead of a `\` when running on windows). But I'll add some tests to make sure we're dealing with the trailing slash thing properly.

Reply:
`URL` is your friend here

```javascript
> (new URL('keys.json', 'https://domain.tld')).toString()
'https://domain.tld/keys.json'
> (new URL('/keys.json', 'https://domain.tld')).toString()
'https://domain.tld/keys.json'
```